convert_checkpoints.py

```diff
@@ -106,82 +105,29 @@
-  updated_weights = {}
-  for key, val in state_dict.items():
-    for qname, qscale_name in _QUANTIZED_WEIGHTS_TO_SCALER_NAME.items():
-      if key.endswith(qname):
-        new_weights, scaler = quantize.quantize_torch_int8(
-            val, reduce_axis=(1,)
-        )
-        updated_weights[key] = new_weights
-        scale_name = key[: -len(qname)] + qscale_name
-        updated_weights[scale_name] = scaler
-  state_dict.update(updated_weights)
-  return state_dict
-
-
-_QUANTIZE_LINEAR_WEIGHTS = {
-    "attention.wq.weight",
-    "attention.wk.weight",
-    "attention.wv.weight",
-    "attention.wo.weight",
-    "feed_forward.w1.weight",
-    "feed_forward.w2.weight",
-    "feed_forward.w3.weight",
-    "output.weight",
+_GEMMA_QUANTIZED_WEIGHTS_TO_SCALER_NAME = {
+    "self_attn.o_proj.weight": "self_attn.o_proj.weight_scaler",
+    "self_attn.wq.weight": "self_attn.wq.weight_scaler",
+    "self_attn.wk.weight": "self_attn.wk.weight_scaler",
+    "self_attn.wv.weight": "self_attn.wv.weight_scaler",
+    "mlp.gate_proj.weight": "mlp.gate_proj.weight_scaler",
+    "mlp.up_proj.weight": "mlp.up_proj.weight_scaler",
+    "mlp.down_proj.weight": "mlp.down_proj.weight_scaler",
+    "embedder.weight": "embedder.weight_scaler",
 }
 
 
-def _quantize_state_dict(state_dict):
+def _quantize_state_dict(state_dict, weight_map, weight_axis):
   updated_weights = {}
   for key, val in state_dict.items():
-    for qname in _QUANTIZE_LINEAR_WEIGHTS:
+    for qname, qscale_name in weight_map.items():
       if key.endswith(qname):
         new_weights, scaler = quantize.quantize_torch_int8(
-            val, reduce_axis=(1,)
+            val, reduce_axis=(weight_axis(key),)
         )
         updated_weights[key] = new_weights
-        scale_name = key + "_scaler"
+        scale_name = key[: -len(qname)] + qscale_name
         updated_weights[scale_name] = scaler.squeeze()
-  tok_weights, tok_scalers = quantize.quantize_torch_int8(
-      state_dict["tok_embeddings.weight"], reduce_axis=(0,)
-  )
-  updated_weights["tok_embeddings.weight"] = tok_weights
-  updated_weights["tok_embeddings.weight_scaler"] = tok_scalers.squeeze()
-  state_dict.update(updated_weights)
-  return state_dict
-
-
-_QUANTIZE_LINEAR_WEIGHTS = {
-    "attention.wq.weight",
-    "attention.wk.weight",
-    "attention.wv.weight",
-    "attention.wo.weight",
-    "feed_forward.w1.weight",
-    "feed_forward.w2.weight",
-    "feed_forward.w3.weight",
-    "output.weight",
-}
-
-
-def _quantize_state_dict(state_dict):
-  updated_weights = {}
-  for key, val in state_dict.items():
-    for qname in _QUANTIZE_LINEAR_WEIGHTS:
-      if key.endswith(qname):
-        new_weights, scaler = quantize.quantize_torch_int8(
-            val, reduce_axis=(1,)
-        )
-        updated_weights[key] = new_weights
-        scale_name = key + "_scaler"
-        updated_weights[scale_name] = scaler.squeeze()
-  tok_weights, tok_scalers = quantize.quantize_torch_int8(
```

Comment:
The embedding is quantized in the existing quantize. Intended to remove it?

Reply:
there was 2 copies of the same function in the file. I removed one